tests/bug-builder/bug-with-width-resizing.test.tsx

```diff
@@ -0,0 +1,48 @@
+import test from "ava"
+import { getTestFixture } from "../fixtures/get-test-fixture"
+
+test("bug with width resizing", async (t) => {
+  const { logSoup, pb } = await getTestFixture(t)
+
+  const soup = await pb
+    .add("board", (board) =>
+      board
+        .setProps({
+          pcbX: 0,
+          pcbY: 0,
+          width: 10,
+          height: 10,
+        })
+        .add("bug", (bug) =>
+          bug.setProps({
+            schPinLabels: {
+              1: "P1",
+              2: "P2",
+              3: "P3",
+              4: "P4",
+              5: "P5",
+              6: "P6",
+              7: "P7",
+              8: "P8",
+            },
+            schPortArrangement: {
+              leftSize: 4,
+              rightSize: 4,
+              schWidth: 2,
```

Comment:
Isnt this supposed the be a top level prop? Why is it inside portArrangement?

Reply:
> this is a pretty good PR, but i think it would be better to modify `getPortArrangementSize` to support taking a `schWidth` prop, transforming things can make the code more complicated over time since it's not clear where the original width is.
>
I thought you meant that I needed to add it there.

Reply:
Here is the bug with schWidth set to 2
![image](https://github.com/user-attachments/assets/a49b309a-1538-45ae-ab57-b57737af9a52)

Reply:
Yea it looks broken, the ports should be outside of the box

Reply:
They are like glued - the box changes its width, but they do not move

![image](https://github.com/user-attachments/assets/204c5d47-2ec0-49d4-81b8-c8ac5ea49168)